### PR TITLE
Allow chargeless firestarters to interact with furniture (#45986)

### LINKED ISSUE #45986: [Magiclysm] Cant fire smoking rack with finger firelighter
# Describe the bug
Cant fire smoking rack with finger firelighter, the game says 'This smoking rack is ready to be fired but you have no fire source'

# Steps To Reproduce

1- Load smoking rack with stuff and fuel
2- Attempt to fire up the rack
3- 'This smoking rack is ready to be fired but you have no fire source'

# Expected behavior
Fire up the rack like with a normal lighter. And a normal lighter does work

# Screenshots

cataclysm-tiles_Qo7Z8jVa7L

Tested it on a save with magiclysm only too, same result

### PR DESCRIPTION
**Pull request by satheon49**
#### Summary

SUMMARY: Bugfixes "Firestarter that could not have charges could not be used to light smoking rack"

#### Purpose of change

Fixes #45986.

#### Describe the solution

Do not check if a firestarter has charges. Check instead whether it could be used to start a fire (with the same logic used for braziers)

#### Describe alternatives you've considered

First solution was to check `!can_have_charges` or `enough charges` but this allowed sunlight dependent firestarters to light indoor smoking racks.

#### Testing

1. Spawn charcoal, chunk of meat and smoking rack
2. Insert food and charcoal
3. Try to light smoking rack.
4. Also test that bifocal glasses do not work indoors but do work outdoors.

#### Additional context
Almost forgot: This also allows those firestarters to work for other activities like cauterizing wounds or using firecrackers and anything else that might use `Character::has_charges( itype_fire, x );` to detect a potential firesource.

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -11149,7 +11149,9 @@ bool Character::has_fire( const int quantity ) const
     } else if( has_item_with_flag( flag_FIRESTARTER ) ) {
         auto firestarters = all_items_with_flag( flag_FIRESTARTER );
         for( auto &i : firestarters ) {
-            if( has_charges( i->typeId(), quantity ) ) {
+            const use_function *usef = i->type->get_use( "firestarter" );
```
Comment (anothersimulacrum) [inline src/character.cpp:11152]: Seems like you need a nullptr check for this.

**Comment by actual-nh** [comment]
This would appear to allow a firestarter that **does** use charges to function for, say, cauterizing, which takes 4 charges (iuse_actor.cpp) - even if it only has 1 charge. (Admittedly, equating charges from one firestarter to another has its difficulties - a book of matches vs a pocket lighter vs something like a butane torch?)

**Comment by satheon49** [comment]
You are right, I will fix it.

**Comment by adamkad1** [comment]
Good job man

**Pull request merged**